### PR TITLE
KFSPTS-21519 making adding and editing a PAAT document through batch job transactional

### PR DESCRIPTION
**Pull request by jhulslander**
I moved code from PayeeACHAccountExtractServiceImpl to the new service PayeeACHAccountDocumentServiceImpl to make  addACHAccount and updateACHAccountIfNecessary.  To avoid adding extra complications to this work, I didn't do "clean code" refactors.  If there are changes you think should be made now, please let me know

### CODE REVIEW

**Comment by jhulslander** [comment]
@cah292 , this is a follow up to my original PR for this user story

**Review by cah292 — CHANGES_REQUESTED** [review]
I noticed a few areas that may need some minor updates, but other than that the changes look good. I'm also okay with deferring more detailed refactoring of the new service to a later PR, given that this PR is largely just moving the affected code from one class to another.

File: src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountDocumentServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+    
+    private ConfigurationService configurationService;
+    private DataDictionaryService dataDictionaryService;
+    private DocumentService documentService;
+    private EmailService emailService;
+    private ParameterService parameterService;
+    private PersonService personService;
+    private SequenceAccessorService sequenceAccessorService;
+
+    @Transactional
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountDocumentServiceImpl.java:61]: ah, I see what you mean, in RassUpdateServiceImpl the annotation is @Transactional(propagation = Propagation.REQUIRES_NEW) ..... is that what I need to do here?

File: src/test/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImplTest.java [hunk 2]
```diff
@@ -555,6 +553,20 @@ private DateTimeService createMockDateTimeService() {
         when(dateTimeService.getCurrentSqlDate()).thenReturn(currentDate);
         return dateTimeService;
     }
+    
+    private PayeeACHAccountDocumentServiceImpl buildPayeeACHAccountDocumentServiceImpl() throws Exception {
+        PayeeACHAccountDocumentServiceImpl payeeACHAccountDocumentService = Mockito.spy(new PayeeACHAccountDocumentServiceImpl());
+        Mockito.doNothing().when(payeeACHAccountDocumentService).addNote(Mockito.any(), Mockito.anyString());
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImplTest.java:559]: I am able to run PayeeACHAccountExtractServiceImplTest locally as it's own test.  Springcontext is not instantiated.  Do you need me to look at something else here?

File: src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountDocumentServiceImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cah292) [reply src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountDocumentServiceImpl.java:61]: Yes, please update the annotations to use the Propagation.REQUIRES_NEW setting. Thanks.

File: src/test/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImplTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (cah292) [reply src/test/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImplTest.java:559]: If the test is working fine when you run it locally, then Mockito.spy() must be properly handling the situation that I thought would have caused issues. Given that, I'm fine with the current state of the rewritten test class. Thanks for checking.

**Comment by jhulslander** [comment]
this is ready for review again

**Review by cah292 — COMMENTED** [review]
Revised changes look good, will merge once the unit tests pass.

**Review by cah292 — APPROVED** [review]
Changes look good, and unit tests pass. Merging.

**Pull request merged**